### PR TITLE
Global Styles: load styles in iframed site editor

### LINKED ISSUE #28581: Global styles CSS-variables don't apply to the iframe in site-editor
## Description
When in the site-editor using an FSE theme, I can see that the CSS-variables get generated properly but they are applied to the main `:root` element. The problem is that they don't get applied to the root element inside the `iframe`, and as a result themes appear broken/unstyled.

## Step-by-step reproduction instructions
1. Install and activate the Q theme from its github repo: https://github.com/aristath/q - The theme.json file has already been updated to accommodate the latest structural changes.
2. Visit the site-editor.

## Expected behaviour
CSS-variables should be applied to the `iframe` and not the root element.

## Actual behaviour
CSS-variables get applied to the root element and the `iframe` does not inherit them.

## WordPress information
- WordPress version: **5.6**
- Gutenberg version: **Master branch** (currently https://github.com/WordPress/gutenberg/tree/502f79a6e0aa65c68562efade58aadc7ab334d4b)
- Are all plugins except Gutenberg deactivated? **Yes**
- Are you using a default theme (e.g. Twenty Twenty-One)? **No**

## Device information
- Device: Desktop
- Operating system: Pop!_OS (basically Ubuntu)
- Browser: Firefox & Chrome

### PR DESCRIPTION
**Pull request by oandregal**
Fixes https://github.com/WordPress/gutenberg/issues/28581

It looks like https://github.com/WordPress/gutenberg/pull/28233 refactored the `useEditorStyles` feature and removed it from the iframe, hence the styles aren't updated upon user changes. This fixes it by bringing it back.

This is what we need to have:

- The Global Styles sidebar needs the preset variables (CSS Custom Properties) in scope in its document, as they are used to power the UI controls values.
- The iframe needs both the preset variables and the block styles.

All of them (presets & block styles) can change upon user changes, so both the iframe and the host document need to be updated upon user changes.

## Test instructions

- Use the TT1-blocks theme.
- Load the site editor and verify that the styles are loaded.
- Change a style in the global sidebar and verify that it's reflected in the editor.

### CODE REVIEW

File: packages/edit-site/src/components/block-editor/index.js [hunk 1]
```diff
@@ -28,16 +28,19 @@ import NavigateToLink from '../navigate-to-link';
...
 	useTypingObserver( body );
+	const iframeRef = useEditorStyles( styles );
 
 	return (
-		<DropZoneProvider>
-			<WritingFlow>
-				<BlockList className="edit-site-block-editor__block-list" />
-			</WritingFlow>
-		</DropZoneProvider>
+		<div ref={ iframeRef }>
```
Comment (nosolosw) [inline packages/edit-site/src/components/block-editor/index.js:37]: I don't think we should add a new div here. Used this to demonstrate the fix. Need to look at how can we attach the ref to a DOM node within the iframe.

**Comment by david-szabo97** [comment]
Looks good! How about putting `useEditorStyles` here:

image

**Comment by david-szabo97** [comment]
Another idea

image

**Comment by aristath** [comment]
Tested and confirmed that it works.
I also tested the alternative from @david-szabo97 in the [comment above](https://github.com/WordPress/gutenberg/pull/28731#issuecomment-773321719) and I can confirm that works as well :+1:

**Comment by nosolosw** [comment]
I was hoping there was a way to attach refs to some DOM node within the iframe but I didn't see how so far. So I went in a different direction: make the iframe updating itself upon style changes ― which was the direction @david-szabo97 pursued at https://github.com/WordPress/gutenberg/pull/28730 but only updating the editor styles.

I don't think this is the final solution but it's working. I can look for a better way to do this in a follow-up PR.

**Review by aristath — APPROVED** [review]

**Comment by sirreal** [comment]
I believe this should this be patched in the 9.9 plugin and backported to the Core beta, so I'm flagging it as such. Feel free to correct.

**Comment by nosolosw** [comment]
> backported to beta

The site editor is not part of the 5.7 beta, hence, I don't think we need to backport this PR.

I don't have a good sense about whether we should release a 9.9.1 with this patch or 10.0. I'll let others with more experience weigh in. As a data point, I can share why I'm undecided so I'm able to learn how others think about this: we've had errors in templates that rendered the edit site unusable that didn't warrant a patch release; also, the site editor is an experimental feature.

File: packages/edit-site/src/components/block-editor/index.js [hunk 2]
```diff
@@ -112,6 +114,7 @@ export default function BlockEditor( { setIsInserterOpen } ) {
 				<Popover.Slot name="block-toolbar" />
 				<Iframe
 					style={ resizedCanvasStyles }
+					editorStyles={ settings.styles }
```
Comment (ellatrix) [inline packages/edit-site/src/components/block-editor/index.js:117]: Why is it also passed further down to `Canvas`?

**Comment by ellatrix** [comment]
@nosolosw Why did you duplicate the logic from `useEditorStyles` instead of using the hook?

**Comment by nosolosw** [comment]
:wave: @ellatrix for context, this is a summary of what needs to happen on the site editor:

- "Global Styles" generates a few things client-side in this editor: block styles (e.g.: `.wp-block-group { color: red }` ) + preset classes (`.has-background-color-pink { background-color: pink }`) + preset values (`:root { --wp--preset--font-size--huge: 42px}`).
- The editor canvas (iframe) needs all of these styles in scope and has to be updated upon user changes. All of those styles can change upon user changes.
- The sidebars (block and global, outside the iframe) need only the preset values in scope and have to be updated upon user changes as well.

I've tried to implement this flow using the hooks and left some comments at https://github.com/WordPress/gutenberg/pull/28731#issuecomment-773480633 I wasn't sure how to make it work, though, hence the hotfix. Plan to iterate on this after doing some digging into the different pieces, happy to work on this with you!

**Pull request merged**